src/rawSHA1_ng_fmt_plug.c

```diff
@@ -330,12 +330,10 @@ static void sha1_fmt_set_key(char *key, int index)
 	vtype  Z   = vsetzero();
 	vtype  X   = vloadu(key);
 	vtype  B;
-#if __AVX512F__ || __MIC__
-	uint32_t len = strlen(key);
-#else
-	uint32_t mask = vmovemask_epi8(vcmpeq_epi8(X, Z));
+
+	// FIXME: even uint64_t won't be long enough for AVX-1024
+	uint64_t mask = vcmpeq_epi8_mask(X, Z);
```

Comment:
Maybe you should benchmark the impact on MIC, and re-add an `#if __MIC__` if needed. Try benchmarking wordlist.c too (eg. just exhausting some huge wordlist without rules against a dummy hash). Perhaps MIC should not use intriniscs at all in wordlist.c (there's a 64-bit alternative below it which is almost as fast).